### DISCUSSION TITLE
memory usage DiskANN vs HNSW

### DISCUSSION THREAD

**Discussion by rere950303** [milvus-io/milvus#35318] [Q&A and General discussion]
When using DiskANN with vector index, how much less memory is used compared to when using HNSW? The vector dimension is 3072.

**Comment by xiaofan-luan** [comment]
For DiskANN, 8GB memory is probably good for 1.5-2m 3072 dimension data.
it also based on the quantization strategy you are using. Can you share a little bit more about your use case

**Reply by rere950303** [reply]
@xiaofan-luan Thank you for your reply. Does 8GB mean the value needed for all querynode replicas, not just a single querynode?

When I tested RAG with DiskANN default settings, I confirmed that `context_precision` and `context_recall` were fine. I think up to 3000 entities will be stored in one segment. Considering memory optimization, are there any recommended DiskANN index build parameters(`PQCodeBugetGBRatio`, `MaxDegree` etc) and search parameters(`search_list`)?

**Reply by xiaofan-luan** [reply]
3000 is a very small number. I recommend to think of using diskann index with 10m data or more.

**Reply by rere950303** [reply]
@xiaofan-luan 3000 is the number in one segment, and the actual data size of the entire collection is much larger. Therefore, considering memory usage, I am going to choose diskann. One thing I am wondering is, in diskann, only the vector index is loaded into memory, and the rest of the scalar fields are on disk?

**Reply by xiaofan-luan** [reply]
see this docuemnt could help https://milvus.io/docs/multi_tenancy.md. You can not control how many data is there in one segment in milvus.

To put all the scalar field on disk, you have to enable mmap feature.

**Reply by rere950303** [reply]
@xiaofan-luan I am planning to use `partition-key based` with multi-tenancy strategy.
If the size of segment is fixed and the size of one entity can be predicted, can't we roughly estimate how many entities will be in the segment?
When you say "It is good to consider diskann for 10m or more", does 10m refer to the data size of the entire collection regardless of how many entities are in one segment?

**Comment by xiaofan-luan** [comment]
all the parameter is related to recall. you can always use default value if you don't have too much idea about how to config except for search_list
